src/m365/aad/commands/m365group/m365group-user-list.ts

```diff
@@ -97,31 +142,65 @@ class AadM365GroupUserListCommand extends GraphCommand {
     }
   }
 
-  private async getOwners(groupId: string, logger: Logger): Promise<User[]> {
+  private async getGroupId(options: Options, logger: Logger): Promise<string> {
+    if (options.groupId) {
+      return options.groupId;
+    }
+
     if (this.verbose) {
-      await logger.logToStderr(`Retrieving owners of the group with id ${groupId}`);
+      await logger.logToStderr('Retrieving Group Id...');
     }
 
-    const endpoint: string = `${this.resource}/v1.0/groups/${groupId}/owners?$select=id,displayName,userPrincipalName,userType`;
+    return await aadGroup.getGroupIdByDisplayName(options.groupDisplayName!);
+  }
 
-    const users = await odata.getAllItems<User>(endpoint);
+  private async getUsers(options: Options, role: string, groupId: string, logger: Logger): Promise<ExtendedUser[]> {
+    const { properties, filter } = options;
 
-    // Currently there is a bug in the Microsoft Graph that returns Owners as
-    // userType 'member'. We therefore update all returned user as owner
-    users.forEach(user => {
-      user.userType = 'Owner';
-    });
+    if (this.verbose) {
+      await logger.logToStderr(`Retrieving ${role} of the group with id ${groupId}`);
+    }
 
-    return users;
-  }
+    const selectProperties: string = properties ?
+      `${properties.split(',').filter(f => f.toLowerCase() !== 'id').concat('id').map(p => p.trim()).join(',')}` :
+      'id,displayName,userPrincipalName,givenName,surname,userType';
+    const allSelectProperties: string[] = selectProperties.split(',');
+    const propertiesWithSlash: string[] = allSelectProperties.filter(item => item.includes('/'));
 
-  private async getMembersAndGuests(groupId: string, logger: Logger): Promise<User[]> {
-    if (this.verbose) {
-      await logger.logToStderr(`Retrieving members of the group with id ${groupId}`);
+    let fieldExpand: string = '';
+    propertiesWithSlash.forEach(p => {
+      if (fieldExpand.length > 0) {
+        fieldExpand += ',';
+      }
+
+      fieldExpand += `${p.split('/')[0]}($select=${p.split('/')[1]})`;
+    });
+
+    const expandParam = fieldExpand.length > 0 ? `&$expand=${fieldExpand}` : '';
+    const selectParam = allSelectProperties.filter(item => !item.includes('/'));
+    const endpoint: string = `${this.resource}/v1.0/groups/${groupId}/${role}/microsoft.graph.user?$select=${selectParam}${expandParam}`;
+
+    let users: ExtendedUser[] = [];
+
+    if (filter) {
+      // While using the filter, we need to specify the ConsistencyLevel header.
+      // Can be refactored when the header is no longer necessary.
+      const requestOptions: CliRequestOptions = {
+        url: `${endpoint}&$filter=${encodeURIComponent(filter)}&$count=true`,
+        headers: {
+          accept: 'application/json;odata.metadata=none',
+          ConsistencyLevel: 'eventual'
+        },
+        responseType: 'json'
+      };
+
+      users = await odata.getAllItems<ExtendedUser>(requestOptions);
```

Comment:
If we're not doing anything with the `users` object we could immediately return the `getAllItems` function result without assigning it to a variable first. The same goes for the `else` clause